examples/v1beta1/pipelineruns/alpha/pipelinerun-array-results-substitution.yaml

```diff
@@ -1,56 +1,95 @@
+                echo -n "[\"$image1\", \"$image2\", \"$image3\"]" | tee $(results.array-results-1.path)
+                echo -n "[\"\", \"$image3\"]" | tee $(results.array-results-2.path)
       - name: task2
         params:
-          - name: foo
-            value: "$(tasks.task1.results.array-results[*])"
-          - name: bar
-            value: "$(tasks.task1.results.array-results[2])"
+          - name: images
+            value: "$(tasks.task1.results.array-results-1[*])"
+          - name: image-1
+            value: "$(tasks.task1.results.array-results-1[2])"
+          - name: image-2
+            value: "$(tasks.task1.results.array-results-2[1])"
         taskSpec:
           params:
-            - name: foo
+            - name: images
               type: array
-              default:
-                - "defaultparam1"
-                - "defaultparam2"
-            - name: bar
+            - name: image-1
+              type: string
+            - name: image-2
               type: string
-              default: "defaultparam1"
           steps:
-            - name: print-foo
+            - name: validate-images
               image: bash:latest
               args: [
-                "echo",
-                "$(params.foo[*])"
+                "$(params.images[*])"
               ]
-            - name: print-bar
+              script: |
+                echo "Validating the length of the array parameter - images"
+                echo "The array parameter, array-param1 must have 3 elements"
+                if [[ $# != 3 ]]; then
+                  exit 1
+                fi
+                echo "Done validating the length of the array parameter - images"
+
+                echo "Validating the first image"
+                if [[ $1 != "abc.dev/sampler/busysample@sha256:19f02276bf8dbdd62f069b922f10c65262cc34b710eea26ff928129a736be791" ]]; then
+                  exit 1
+                fi
+                echo "Done validating the first image"
+
+                echo "Validating the second image"
+                if [[ $2 != "ubuntu" ]]; then
+                  exit 1
+                fi
+                echo "Done validating the second image"
+
+                echo "Validating the third image"
+                if [[ $3 != "xyz.dev/awesomeness/awesomeness" ]]; then
+                  exit 1
+                fi
+                echo "Done validating the third image"
+            - name: validate-image
               image: ubuntu
               script: |
                 #!/bin/bash
-                VALUE=$(params.bar)
-                EXPECTED=3
+                VALUE=$(params.image-1)
+                EXPECTED="xyz.dev/awesomeness/awesomeness"
                 diff=$(diff <(printf "%s\n" "${VALUE[@]}") <(printf "%s\n" "${EXPECTED[@]}"))
                 if [[ -z "$diff" ]]; then
-                    echo "Get expected: ${VALUE}"
+                    echo "Got expected: ${VALUE}"
```

Comment:
I think we may need to remove the exit 0 here. Otherwise the next validation won't happen

Reply:
yup, you are absolutely right 💯 Thanks for pointing that out 👍 Dropped `exit 0` from here.